### PR TITLE
introduce replay example

### PR DESCRIPTION
**Pull request by dzinad**
Refs #51.

### CODE REVIEW

**Comment by dzinad** [comment]
Could you please have a look at coroutines usage in HistoryFilesViewController? I've changed the approach a bit.

File: app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var mapboxReplayer: MapboxReplayer
+    private lateinit var locationComponent: LocationComponentPlugin
+    private lateinit var navigationCamera: NavigationCamera
+    private lateinit var viewportDataSource: MapboxNavigationViewportDataSource
+    private lateinit var binding: ActivityReplayHistoryLayoutBinding
+    private lateinit var activityResultLauncher: ActivityResultLauncher<Intent>
+    private var isLocationInitialized = false
+    private val pixelDensity = Resources.getSystem().displayMetrics.density
+    private val overviewPadding: EdgeInsets by lazy {
```
Comment (RingerJK) [inline app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt:67]: let's simplify a bit example: remove `EdgeInsets`s and lock the screen to portrait only. This way we remove code that is not directly related to `Replay`

File: app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+        viewportDataSource = MapboxNavigationViewportDataSource(
+            binding.mapView.getMapboxMap()
+        )
+        val mapboxMap = binding.mapView.getMapboxMap()
+        navigationCamera = NavigationCamera(
+            mapboxMap,
+            binding.mapView.camera,
+            viewportDataSource
+        )
+        initialCameraOptions?.let { mapboxMap.setCamera(it) }
```
Comment (RingerJK) [inline app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt:173]:
```suggestion
       mapboxMap.setCamera(
         CameraOptions.Builder()
          .zoom(DEFAULT_INITIAL_ZOOM)
          .build()
       )
```

File: app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+            30.0 * pixelDensity,
+            380.0 * pixelDensity,
+            110.0 * pixelDensity,
+            40.0 * pixelDensity
+        )
+    }
+
+    private val initialCameraOptions: CameraOptions? = CameraOptions.Builder()
+        .zoom(DEFAULT_INITIAL_ZOOM)
+        .build()
```
Comment (RingerJK) [inline app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt:102]:
```suggestion
```

File: app/src/main/java/com/mapbox/navigation/examples/replay/HistoryFilesViewController.kt [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        }
+    }
+
+    fun requestHistoryFiles(context: Context, connectionCallback: (Boolean) -> Unit) {
+        requestHistory(context, connectionCallback)
+    }
+
+    private fun requestHistory(context: Context, connectionCallback: (Boolean) -> Unit): Job {
+        return lifecycleScope.launch {
+            val drives = historyFilesApi.requestHistory().toMutableList()
```
Comment (RingerJK) [inline app/src/main/java/com/mapbox/navigation/examples/replay/HistoryFilesViewController.kt:52]: let's also put to the drives list labels for each category, smth like (first that I found 🎸 ):
image

File: app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+            },
+            object : OnMapLoadErrorListener {
+                override fun onMapLoadError(eventData: MapLoadingErrorEventData) {
+                    // intentionally blank
+                }
+            }
+        )
+    }
+
+    private val locationObserver = object : LocationObserver {
```
Comment (RingerJK) [inline app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt:198]: let's put `val`s on top of functions in a class' hierarchy

File: app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+        initNavigation()
+        handleHistoryFileSelected()
+        initMapStyle()
+        activityResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            if (it.resultCode == HistoryFilesActivity.REQUEST_CODE) {
+                handleHistoryFileSelected()
+            }
+        }
+
+        binding.selectHistoryButton.setOnClickListener {
```
[the remark this answers is not in the document]
Comment (dzinad) [reply app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt:160]: Please have a close look at this point. Wanna be sure that I used start/stop/reset methods correctly.

File: app/src/main/java/com/mapbox/navigation/examples/replay/HistoryFilesViewController.kt (hunk 4 above)
Comment (dzinad) [reply app/src/main/java/com/mapbox/navigation/examples/replay/HistoryFilesViewController.kt]: Added sections to the RecyclerView.

File: app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (RingerJK) [reply app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt:160]: yeah, that should be enough 
```
mapboxReplayer.clearEvents()
mapboxNavigation.stopTripSession()
```

> reset methods correctly

what do you mean `reset`?

**Comment by RingerJK** [comment]
looks like we need camera's paddings for portrait mode 

https://user-images.githubusercontent.com/8738058/173861924-fedebda6-0ca1-4088-98e8-583ad9a87999.mp4

**Review by abhishek1508 — APPROVED** [review]
